Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/DCAuthManager.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.microsoft.azuretools.authmanage;
+
+import com.microsoft.aad.adal4j.AuthenticationCallback;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.azuretools.adauth.AuthContext;
+import com.microsoft.azuretools.adauth.AuthResult;
+import java.io.IOException;
+
+public class DCAuthManager extends BaseADAuthManager {
+    private static DCAuthManager instance = null;
+
+    public static DCAuthManager getInstance() throws IOException {
```

Comment:
Or to use an inner class as singleton holder, ensuring both a) lazy load, b) better efficiency

Reply:
Use an inner class to do this.
```
private static class LazyLoader {
    static final DCAuthManager INSTANCE = new DCAuthManager();
}

public static DCAuthManager getInstance() {
    return LazyLoader.INSTANCE;
}
```

When creating the auth manager, it will check the Azure environment. If it not correct, an `IOException` will be thrown. But this is not allowed when initiating a static variable, so I will use `ExceptionInInitializerError` in the base constructor instead.
https://docs.oracle.com/javase/6/docs/api/java/lang/ExceptionInInitializerError.html